### ISSUE TITLE
[Bug] Wildcards in sitelist prevents the toggle function from working as intended.

### ISSUE THREAD

**Issue by XandaPanda42** [darkreader/darkreader#13134]
### Prerequisites

- [X] I [searched for any existing report](https://github.com/darkreader/darkreader/issues?q=is%3Aissue) about this bug to avoid opening a duplicate.
- [X] I can reproduce this bug in a new, unmodified web browser profile with Dark Reader installed as the only extension.
- [X] I understand I need to use the [Broken Website Report template](https://github.com/darkreader/darkreader/issues/new?assignees=&labels=Broken+Website&template=broken-website-report.yml&title=%5BBroken+Website%5D+Replace+with+title) if this bug I am reporting occurs on a single website.

### Bug Description

I've got it set to be dark mode by default. If I use a wildcard (unsure if that's the right term) on a domain name it works as expected, making all subdomains lightmode, but attempting to toggle darkmode on a specific subdomain doesn't toggle it as I assumed it would. 

For example:

In a site list with only "en.wikipedia.org" in it, it overwrites the default, and shows the page in lightmode. Toggling darkmode (using the context menu or the extension dialogue) while on that page removes "en.wikipedia.org" from the list and reverts the page to default darkmode as normal.

In a site list with only "*.wikipedia.org" in it, shows "en.wikipedia.org" in lightmode as expected. Normally this is expected behavior, but since the site is already in lightmode, Telling the extension to toggle does nothing, except add the subdomain explicitly to the site list, keeping the site in lightmode.

### Website Address

All

### Steps To Reproduce

1. Setup the extension to set pages to dark mode by default. Using the "Not Invert Listed" list.

2. Add "*.wikipedia.org" or put a wildcard in front of any domain and add it to your site list.

3. Navigate to "wikipedia.org" or "simple.wikipedia.org" or any subdomain of your chosen site and attempt to toggle the site into darkmode using either the context menu or the toolbar.

### Expected Behavior

1. The site toggles between dark and lightmode as expected.

or

2. The site shows a popup alerting the user that toggling is not supported when using wildcards this way.

or

3. The button to toggle the site is greyed out to let the user know it's not supported using this method.

### Actual Behavior

Instead of toggling the site back into dark mode as expected, the extension adds the subdomain to the site list. But since the site is already in the list, it doesn't affect the appearance of the page. It doesn't alert the user that anything is wrong using the UI or a log file.

### Screenshots

en.wikipedia.org default.
![en-wikipedia-default](https://github.com/user-attachments/assets/6065adf8-68b6-4ebb-a087-9c40c7d5c210)

en.wikipedia.org added to the site list.
![en-wikipedia-sitelist](https://github.com/user-attachments/assets/78f6dce8-00a8-4310-8aee-b51430d4777f)

simple.wikipedia.org default.
![simple-wikipedia-default](https://github.com/user-attachments/assets/72253396-1f7e-49bb-8267-44729e3614cb)

simple.wikipedia.org added to the site list.
![simple-wikipedia-sitelist](https://github.com/user-attachments/assets/445f353d-7903-4e89-8e57-17c5ae979cc7)

Attempting to toggle "simple.wikipedia.org" while "*.wikipedia.org" is in the site list.
[Screencast from 2024-08-14 20-17-36.webm](https://github.com/user-attachments/assets/3bea5d68-6f12-4a4a-83dd-1564d6ce1b16)

### Operating System

Ubuntu

### Web Browser name and version

Firefox, 124.0.1

### Dark Reader version

4.9.88

### Additional Context

I am aware that other methods like expressions I believe are available, and I will look into and use them going forward, but this report is mainly about the extension not allowing toggling while using a wildcard in a fairly standard way, then making changes to the config, because it assumes it is working as intended, and the only feedback it gives is: "This button isn't doing anything"

**Comment by Myshor** [comment]
Yeah... the lists are not prepared to mess with them using `*.`  so expected behavior 1 will be hard to implement as we can not remove wildcard that easy.

2 or 3 is more possible - I don't know how it is implemented in the code to check if site is included on the list or not so it is added or removed from the list but adding check if there site is restricted as wildcard should not be hard.

Calling @alexanderby as it will need some fixes in the code. 😉
